oasis-build/src/rpc.rs

```diff
@@ -433,15 +446,41 @@ fn convert_type_def(
         .data
         .to_string();
     if def.is_enum() {
-        if !def.is_payloadfree() {
-            // TODO: convert Rust struct enum to tagged union
-            return Err(UnsupportedTypeError::ComplexEnum(
-                tcx.def_span(def.did).into(),
-            ));
-        }
+        let variants = def
```

Comment:
I'm sure it works but do you think you could add a couple of tests for this part?

Reply:
I can't unit test these because the types are created by the compiler and don't have public creator methods. The `:/tests` directory is full of integration tests, though.

Reply:
got it. I have to get more familiar with oasis-rs :)